packages/shared/lib/services/sync/data/records.service.ts

```diff
@@ -440,19 +441,27 @@ export async function getAllDataRecords(
             `)
         );
 
-        const result = encryptionManager.decryptDataRecords(rawResult, 'record') as unknown as SyncDataRecord[];
-
         if (result.length === 0) {
             return { success: true, error: null, response: { records: [], next_cursor: nextCursor } };
         }
 
-        const customerResult = result.map((item) => item.record);
+        const customerResult = result.map((item: RawDataRecordResult) => {
+            const decryptedRecord = encryptionManager.decryptDataRecord(item);
+            if (!decryptedRecord) {
+                return decryptedRecord;
+            }
+            const nextCursor = item.record._nango_metadata.last_modified_at.toString();
+            const id = item.id;
+            const encodedCursorValue = Buffer.from(`${nextCursor}||${id}`).toString('base64');
```

Comment:
nitpick: delimiter can be a single char. That's always one char that won't be needed to be sent over the network :)